src/render.rs

```diff
@@ -31,35 +38,7 @@ pub enum Content<'t, 'py> {
     Int(BigInt),
 }
 
-#[derive(Error, Debug, Diagnostic, PartialEq, Eq)]
-pub enum RenderError {
-    #[error("Failed lookup for key [{key}] in {object}")]
-    VariableDoesNotExist {
-        key: String,
-        object: String,
-        #[label("key")]
-        key_at: SourceSpan,
-        #[label("{object}")]
-        object_at: Option<SourceSpan>,
-    },
-}
-
-#[derive(Error, Debug)]
-pub enum PyRenderError {
-    #[error(transparent)]
-    PyErr(#[from] PyErr),
-    #[error(transparent)]
-    RenderError(#[from] RenderError),
-}
-
-impl PyRenderError {
-    pub fn try_into_render_error(self) -> Result<RenderError, PyErr> {
-        match self {
-            Self::RenderError(err) => Ok(err),
-            Self::PyErr(err) => Err(err),
-        }
-    }
-}
+pub type TemplateResult<'t, 'py> = Result<Option<Content<'t, 'py>>, PyRenderError>;
```

Comment:
If you're comfortable with an interactive rebase, I think introducing `TemplateResult` would make a nice standalone commit before the rest of the refactor.

Not critical if it's too much hassle.

Reply:
If you do this, I think it would also be neater if code moves were in their own commits instead of mixed with refactoring. (I.e. refactor within a file as one commit, then move it to a new file as a second commit.)

Again this is a nice to have if not too big a pain.

Reply:
`TemplateResult` got it's own commit, and I agree that having the move could have been nice in its own commit. But seems like the payoff for the effort was a little to small this time, especially since it's nice to have each commit properly compile, which all commits now do.